app/assets/javascripts/galleries/add_new.js

```diff
@@ -86,8 +86,9 @@ function bindFileInput(fileInput) {
         submitButton.prop('disabled', false);
 
         // extract key and generate URL from response
+        // replace https://glowfic-constellation.s3.amazonaws.com with Cloudfront URL
         var key   = $(data.jqXHR.responseXML).find("Key").text();
-        var url   = 'https://d1anwqy6ci9o1i.cloudfront.net/' + key;
+        var url   = $(data.jqXHR.responseXML).find("Location").text().replace(form.data('url'), 'https://d1anwqy6ci9o1i.cloudfront.net');
```

Comment:
Tested locally and got `https://d1anwqy6ci9o1i.cloudfront.net/users%2F1%2Ficons%2F78f49a9a-c161-4395-9ea5-f8ab4df7238e_aestrix.jpeg` instead of `https://d1anwqy6ci9o1i.cloudfront.net/users/1/icons/78f49a9a-c161-4395-9ea5-f8ab4df7238e_aestrix.jpeg`, seems to be over-escaping?